v2/internal/reconcilers/entra/entra_securitygroup_reconciler.go

```diff
@@ -0,0 +1,499 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package entra
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+
+	"github.com/go-logr/logr"
+	msgraphsdkgo "github.com/microsoftgraph/msgraph-sdk-go"
+	"github.com/microsoftgraph/msgraph-sdk-go/groups"
+	msgraphmodels "github.com/microsoftgraph/msgraph-sdk-go/models"
+	"github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
+	"github.com/rotisserie/eris"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	asoentra "github.com/Azure/azure-service-operator/v2/api/entra/v1"
+	"github.com/Azure/azure-service-operator/v2/internal/config"
+	"github.com/Azure/azure-service-operator/v2/internal/identity"
+	"github.com/Azure/azure-service-operator/v2/internal/reconcilers"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/conditions"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+)
+
+// EntraSecurityGroupReconciler reconciles an Entra security group.
+// TODO: Factor out common code shared with other Entra resources into entraGenericReconciler
+type EntraSecurityGroupReconciler struct {
+	reconcilers.ReconcilerCommon
+	ResourceResolver   *resolver.Resolver
+	CredentialProvider identity.CredentialProvider
+	Config             config.Values
+	EntraClientFactory EntraConnectionFactory
+}
+
+var _ genruntime.Reconciler = &EntraSecurityGroupReconciler{}
+
+func NewEntraSecurityGroupReconciler(
+	kubeClient kubeclient.Client,
+	entraClientFactory EntraConnectionFactory,
+	resourceResolver *resolver.Resolver,
+	positiveConditions *conditions.PositiveConditionBuilder,
+	cfg config.Values,
+) *EntraSecurityGroupReconciler {
+	return &EntraSecurityGroupReconciler{
+		ResourceResolver:   resourceResolver,
+		Config:             cfg,
+		EntraClientFactory: entraClientFactory,
+		ReconcilerCommon: reconcilers.ReconcilerCommon{
+			KubeClient:         kubeClient,
+			PositiveConditions: positiveConditions,
+		},
+	}
+}
+
+func (r *EntraSecurityGroupReconciler) CreateOrUpdate(
+	ctx context.Context,
+	log logr.Logger,
+	eventRecorder record.EventRecorder,
+	obj genruntime.MetaObject,
+) (ctrl.Result, error) {
+	group, err := r.asSecurityGroup(obj)
+	if err != nil {
+		return ctrl.Result{}, eris.Wrapf(err, "creating or updating security group %s", group.Name)
+	}
+
+	// TODO: Respect annotation serviceoperator.azure.com/reconcile-policy
```

Comment:
Should probably be done as part of this PR?

Reply:
Oh wait, you shouldn't actually need to do anything here - handling of this is taken care of in the `reconcilers/generic/generic_reconciler` -- see `handleSkipReconcile`. I was thinking it needed to be handled in each and every impl, but now that I am thinking about it I believe we refactored that so it was taken care of at the upper level.